src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -149,7 +160,13 @@ public RestClientMethod BuildMethod(Operation operation, HttpRequest httpRequest
                 .Where(rp => !IsIgnoredHeaderParameter(rp))
                 .ToArray();
 
-            return parameters.ToDictionary(rp => rp, requestParameter => BuildParameter(requestParameter));
+            var result = new Dictionary<RequestParameter, Parameter>();
+            foreach (var parameter in parameters)
+            {
+                if (!result.ContainsKey(parameter))
```

Comment:
It is not normal to have one and the same parameter in the method more than once. It may be, however, a side effect of incorrectly transformed swagger. We should fail here rather than swallow it.